### PR TITLE
ddl: write binlog when ddl job state is done.

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
Old TiDB writes DDL binlog in the transaction of job.state: done -> synced. The problem here is if a DML operation between the public to synced, the DML may succeed in TiDB, but failed after binlog synced, Because the DDL binlog has not been writed.

### What is changed and how it works?
This PR change to: write ddl binlog in the transaction of job.state: other -> done.

### Check List 

Tests 

Code changes

 - Has exported function/method change

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by crazycs520** [comment]
@zimulala @WangXiangUSTC  PTAL

File: ddl/ddl_worker.go [hunk 1]
```diff
@@ -398,7 +397,13 @@ func (w *worker) handleDDLJobQueue(d *ddlCtx) error {
 				return errors.Trace(err)
 			}
 			err = w.updateDDLJob(t, job, runJobErr != nil)
-			return errors.Trace(w.handleUpdateJobError(t, job, err))
+			if err = w.handleUpdateJobError(t, job, err); err != nil {
+				return errors.Trace(err)
+			}
+			if job.IsDone() || job.IsRollbackDone() {
+				binloginfo.SetDDLBinlog(d.binlogCli, txn, job.ID, job.Query)
```
[the remark this answers is not in the document]
Comment (july2993) [reply ddl/ddl_worker.go]: see the discussion at https://github.com/pingcap/tidb/issues/9304

**Pull request merged**